Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -108,6 +136,46 @@ public void Serialize(IIonWriter writer, object item)
             writer.StepOut();
         }
 
+        private object InvokeIonConstructor(ConstructorInfo ionConstructor, IIonReader reader)
+        {
+            var parameters = ionConstructor.GetParameters();
+            var arguments = new object[parameters.Length];
+
+            if (parameters.Length > 0)
+            {
+                // Compute mapping between parameter names and index in parameter array.
+                var paramIndexMap = new Dictionary<string, int>();
+                for (int i = 0; i < parameters.Length; i++)
+                {
+                    var param = (IonPropertyName)parameters[i].GetCustomAttribute(typeof(IonPropertyName));
+                    if (param != null)
+                    {
+                        paramIndexMap.Add(param.Name, i);
+                    }
+                    else
```

Comment:
Can you change this so it's a param == null -> throw, and the addition to the map occurs after and not within a nested block? It's a bit cleaner that way.

Reply:
Addressed.